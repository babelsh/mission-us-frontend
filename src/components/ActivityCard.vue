<template>
  <v-card>
    <v-img
      :height="imageHeight"
      :src="imgSrc"
    ></v-img>
    <v-card-title>
      <div>
        <span class="title">{{ title }}</span>
      </div>
    </v-card-title>
    <v-card-actions style="padding-top: 0px">
      <v-btn flat>
        Saber mais
      </v-btn>
      <v-spacer></v-spacer>
      <v-btn color="secondary">
        Feito
      </v-btn>
    </v-card-actions>
  </v-card>
</template>

<script>

export default {
  name: 'activity-card',
  props: {
    description: {
      type: String,
      required: true,
    },
    imgSrc: {
      type: String,
      required: true,
    },
    title: {
      type: String,
      required: true,
    },
  },
  computed: {
    imageHeight() {
      switch (this.$vuetify.breakpoint.name) {
        case 'xs': return '150px';
        case 'sm': return '200px';
        case 'md': return '200px';
        case 'lg': return '200px';
        case 'xl': return '200px';
        default: console.log('inv√°lid screen size.');
      }

      return '';
    },
  },
};
</script>
