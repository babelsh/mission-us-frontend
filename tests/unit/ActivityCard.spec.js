import { mount } from '@vue/test-utils';
import ActivityCard from '@/components/ActivityCard.vue';
import Vue from 'vue';
import Vuetify from 'vuetify';

describe('ActivityCard', () => {
  let wrapper;

  beforeEach(() => {
    Vue.use(Vuetify);
  });

  it('ActivityCard works with property', () => {
    wrapper = mount(ActivityCard, {
      propsData: {
        title: 'card title',
        imgSrc: '',
        description: '',
      },
    });
    expect(wrapper.props().title).toBe('card title');
  });

  it('renders the component properly', () => {
    wrapper = mount(ActivityCard, {
      propsData: {
        title: '',
        imgSrc: '',
        description: '',
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
