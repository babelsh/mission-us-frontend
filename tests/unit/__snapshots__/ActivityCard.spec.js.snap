// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityCard renders the component properly 1`] = `
<div class="container pa-0">
  <div class="v-card theme--light">
    <div class="v-responsive v-image" style="height: 200px;">
      <div class="v-responsive__content"></div>
    </div>
    <div class="v-card__title no-padding-bottom">
      <div class="text-truncate">
        <p class="title text-truncate">

        </p>
        <p class="body-2 grey--text text-truncate">

        </p>
      </div>
    </div>
    <div class="v-card__actions no-padding-top"><button type="button" class="v-btn v-btn--flat theme--light">
        <div class="v-btn__content">
          Saber mais
        </div>
      </button>
      <div class="spacer"></div> <button type="button" class="v-btn theme--light primary">
        <div class="v-btn__content">
          Feito
        </div>
      </button>
    </div>
  </div>
</div>
`;
